bukuserver/views.py

```diff
@@ -46,8 +46,11 @@ def index(self):
     def search(self):
         "redirect to bookmark search"
         form = forms.HomeForm()
-        buku_filter = bs_filters.BookmarkBukuFilter(deep=form.deep.data, regex=form.regex.data)
-        url = url_for('bookmark.index_view', **{filter_key(buku_filter): form.keyword.data})
+        regex, markers = form.regex.data, form.markers.data
+        deep, all_keywords = (x and not regex for x in [form.deep.data, form.all_keywords.data])
+        flt = bs_filters.BookmarkBukuFilter(deep=deep, regex=regex, markers=markers, all_keywords=all_keywords)
+        vals = ([('', form.keyword.data)] if not markers else enumerate(buku.split_by_marker(form.keyword.data)))
+        url = url_for('bookmark.index_view', **{filter_key(flt, idx): val for idx, val in vals})
```

Comment:
Old behaviour without `markers`, otherwise splitting into multiple keywords based on detected markers.

(…Also I removed combinations of `regex` with `deep`/`all_keywords` since those do nothing :sweat_smile:)